### PR TITLE
[FIX] core: install new dependencies of module in 'to upgrade' state

### PR DESCRIPTION
**Pull request by sbidoul**
When a custom module is in 'to upgrade' state, and the new version has a new dependency that is not yet installed, Odoo refuses to upgrade it, and says the new dependency is unmet.

This commit fixes this by calling button_upgrade() in this situation too, so as to recursively mark all dependencies for upgrade/installation.

This situation arises in a version migration scenarios. Custom modules are in 'to upgrade' state when the database comes back from the Odoo migration service. If one of these custom modules has a new dependency after migration, it refuses to upgrade.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by sbidoul** [comment]
This is only a partial fix for #72703.

**Review by rco-odoo — APPROVED** [review]
Looks okay to me.  Waiting for a final review by @Elkasitu.

**Review by Elkasitu — APPROVED** [review]
LGTM, doesn't seem to break any install/uninstall tests
robodoo r+

**Pull request closed without merging**
